src/components/SidePanelControlBar.tsx

```diff
@@ -32,11 +32,19 @@ const SearchBarComponent: FC<SidePanelControlBarProps> = ({
   const savedRef = useRef<HTMLButtonElement | null>(null);
   const { dispatch, appFilter } = useFilter();
 
-  let filterCount = Object.keys(appFilter).length;
-
-  if (shouldFilterSavedProperties) {
-    // Exclude opa_id from filterCount, which counts opa_id as a filter by default
-    filterCount--;
+  const filterCount = () => {
```

Comment:
With the rewritten `filterCount`, the UI is displaying the correct count now. However, `filterCount` is now a function, and it's being called multiple times whenever the filter screen loads, or when a user clicks one of the filter buttons.